File: bit-manipulation/add-binary/add-binary.ts
function addBinary(a: string, b: string): string {
    let i = a.length - 1, j = b.length - 1;
    let carry = 0;
    let result: string[] = [];

    while (i >= 0 || j >= 0 || carry > 0) {
        let sum = carry;
        if (i >= 0) sum += +a[i--];   // символ -> число
        if (j >= 0) sum += +b[j--];

        result.push((sum % 2).toString());
        carry = Math.floor(sum / 2);
    }

    return result.reverse().join("");
};

File: bit-manipulation/add-binary/add-binary.cs
public class Solution {
    public string AddBinary(string a, string b) {
        int i = a.Length - 1, j = b.Length - 1, carry = 0;
        var result = new StringBuilder();

        while (i >= 0 || j >= 0 || carry > 0) {
            int sum = carry;
            if (i >= 0) sum += a[i--] - '0';
            if (j >= 0) sum += b[j--] - '0';

            result.Insert(0, (sum % 2).ToString());
            carry = sum / 2;
        }

        return result.ToString();
    }
}